impurityModel/ed/remove.py

```diff
@@ -55,37 +55,41 @@ def utuple(i, state):
         return (), 0
 
 
-def uint(n_spin_orbitals, i, state):
+def uint(n_spin_orbitals: int, i: int, state: int) -> tuple[int, int]:
     """
     Remove electron at orbital i in state.
 
     Parameters
     ----------
-    n_spin_orbitals : int
+    n_spin_orbitals:
         Total number of spin-orbitals in the system.
-    i : int
+    i:
         Spin-orbital index
-    state : int
+    state:
         Product state.
 
     Returns
     -------
-    state_new : int
+    state_new:
         Product state
-    amp : int
+    amp:
         Amplitude. 0, -1 or 1.
 
     """
-    # String representation of product state.
-    s = psr.int2str(state, n_spin_orbitals)
-    if s[i] == "0":
-        return -1, 0
-    elif s[i] == "1":
-        state_new = state - 2 ** (n_spin_orbitals - i - 1)
-        amp = 1 if s[:i].count("1") % 2 == 0 else -1
+    i_right = n_spin_orbitals - 1 - i
+    if state & (1 << i_right):
+        # Spin-orbital is occupied
+        # Remove electron with XOR-operator
+        state_new = state ^ (1 << i_right)
```

Comment:
Using `& ~` instead of `^` here did not make a big difference in speed. 
```
In [24]: %timeit -n 10000_000 uint(7,5, 18)
446 ns ± 20 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)

In [25]: %timeit -n 10000_000 uint_test(7,5, 18)
454 ns ± 7.92 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)
```
`uint` removes using `^` and  `uint_test` removes using `& ~`.

Reply:
changed to `&~` and added a comment that `^` also works fine.